File: src/CSharp/162.FindPeakElement.cs

using System;
using System.Text;
using System.Collections.Generic;

namespace PeefyLeetCode.FindPeakElement
{
public class Solution {
    public int FindPeakElement(int[] nums) {
        int l = 0, r = nums.Length - 1;
        while (l < r) {
            int mid = (l + r) / 2;
            if (nums[mid] > nums[mid + 1])
                r = mid;
            else
                l = mid + 1;
        }
        return l;
    }
}
}

File: src/Python/101-200/162.FindPeakElement.py

class Solution:
    def findPeakElement(self, nums):
        l = 0
        right = len(nums) - 1
        while l < r:
            mid = (l + r) // 2
            if nums[mid] > nums[mid + 1]:
                r = mid
            else:
                l = mid + 1
        return l


if __name__ == "__main__":
    solution = Solution()
    print(solution.findPeakElement([1,2,3,1]))
    print(solution.findPeakElement([1,2,1,3,5,6,4]))
